docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+---
+title: The Journey of developing the MUI Masonry component
+description: Benny Joo recaps the development process of the MUI Masonry component.
+date: 2021-09-17T00:00:00.000Z
+authors: ['bennyjoo"]
+card: true
+---
+
+## Request from the community
+
+Just over two years ago, a [request](https://github.com/mui-org/material-ui/issues/17000) for a masonry component was opened. Upvoted by around 200 MUI users, this was one of the most popular component requests so far.
+
+![Screenshot 2021-09-15 at 09 13 33](https://user-images.githubusercontent.com/32841130/133396518-74c7792e-e4bc-403d-aad4-2585a7876206.png)
+
+According to the 70 responses to our survey, we learned that, in a masonry, MUI users want to:
+
+- display less than 50 items,
+- span multiple columns,
+- insert each item in the shortest column.
+
+![Screenshot 2021-09-15 at 09 29 54](https://user-images.githubusercontent.com/32841130/133398931-a9ee1aa8-11ca-47e7-a356-d051dab30435.png)
+
+![Screenshot 2021-09-15 at 09 29 59](https://user-images.githubusercontent.com/32841130/133398945-0e3ea284-8d67-47a3-afad-36a95a6f6925.png)
+
+![Screenshot 2021-09-15 at 09 30 06](https://user-images.githubusercontent.com/32841130/133398964-059e0faf-d624-472d-8fbb-9b1ede41cce6.png)
+
+With the help of these survey results, we came up with the following requirements:
+
+**Must-have**:
+
+- Items can have arbitrary heights.
+- Items should flow horizontally.
+- Each item is pushed to the shortest column.
+- High performance
+
+**Should-have**:
+
+- The number of columns, `columns`, and the spacing between items, `spacing`, can be passed into the component.
+- Responsive number of columns (breakpoints)
+- Responsive spacing (breakpoints)
+
+**Nice-to-have**:
+
+- Server side rendering
+- Span multiple columns in a masonry
+- Logical tab order
+- Support for lazy loading/virtualisation
+
+In short, we wanted to build a masonry component that is light-weight enough to be performant and easy-to-use, but solid enough to do its job.
+
+## Two possible solutions: CSS Grid or CSS Flexbox?
+
+Since our goal was to develop a light-weight, performant masonry component, an absolute positioning implementation, which most existing masonry libraries make use of, was not an option. It requires too much JavaScript, and therefore it is likely to be heavy-weight.
+
+This left either CSS Grid or CSS Flexbox, but each came with its own advantages and disadvantages.
+
+**1. CSS Grid + Configuring `grid-row-start` or `grid-row-end`of items**
+
+The idea is to create a grid container, `<Masonry />`, that contains grid items, `<MasonryItem />`s, each of which contains a child. Then, we compute the height of each child and set the height of its wrapper, `<MasonryItem />`, to this value. This allows every `<MasonryItem />` to be able to have a different height and to be large enough to display its child.
```

Comment:
AFAIK, we never use the `<`, `>` when we mention the components in the docs.

```suggestion
The idea is to create a grid container, `Masonry`, that contains grid items, `<MasonryItem />`s, each of which contains a child. Then, we compute the height of each child and set the height of its wrapper, `<MasonryItem />`, to this value. This allows every `<MasonryItem />` to be able to have a different height and to be large enough to display its child.
```

x the other cases and the docs: https://mui.com/components/masonry/